### PR TITLE
Update to Liquigraph 4.0.1

### PR DESCRIPTION
**Pull request by fbiville**
Requires JDK 11

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/liquigraph.rb [hunk 1]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
   end
 
   depends_on "maven" => :build
-  depends_on :java => "1.8"
+  depends_on :java => "11"
```
Comment (bayandin) [inline Formula/liquigraph.rb:16]:
```suggestion
  depends_on "openjdk@11"
```

File: Formula/liquigraph.rb [hunk 2]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
...
 
   def install
-    cmd = Language::Java.java_home_cmd("1.8")
+    cmd = Language::Java.java_home_cmd("11")
     ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
```
Comment (bayandin) [inline Formula/liquigraph.rb:20]:
```suggestion
    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
```

File: Formula/liquigraph.rb [hunk 3]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
...
     system "mvn", "-B", "-q", "-am", "-pl", "liquigraph-cli", "clean", "package", "-DskipTests"
     (buildpath/"binaries").mkpath
     system "tar", "xzf", "liquigraph-cli/target/liquigraph-cli-bin.tar.gz", "-C", "binaries"
     libexec.install "binaries/liquigraph-cli/liquigraph.sh"
     libexec.install "binaries/liquigraph-cli/liquigraph-cli.jar"
-    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", Language::Java.java_home_env("1.8")
+    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", Language::Java.java_home_env("11")
```
Comment (bayandin) [inline Formula/liquigraph.rb:26]:
```suggestion
    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", :JAVA_HOME => Formula["openjdk@11"].opt_prefix
```

**Review by bayandin — CHANGES_REQUESTED** [review]
Can we use the latest OpenJDK (currently it's 13 in Homebrew)?

**Comment by fbiville** [comment]
> Can we use the latest OpenJDK (currently it's 13 in Homebrew)?

11 is a minimum, but it should work with 13.

File: Formula/liquigraph.rb [hunk 4]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
   end
 
   depends_on "maven" => :build
-  depends_on :java => "1.8"
+  depends_on "openjdk@11"
```
Comment (SMillerDev) [inline Formula/liquigraph.rb:16]:
```suggestion
  depends_on "openjdk"
```

File: Formula/liquigraph.rb [hunk 5]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
...
 
   def install
-    cmd = Language::Java.java_home_cmd("1.8")
-    ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
+    cmd = Language::Java.java_home_cmd("11")
+    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
```
Comment (SMillerDev) [inline Formula/liquigraph.rb:20]:
```suggestion
    ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
```

File: Formula/liquigraph.rb [hunk 6]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
...
     system "mvn", "-B", "-q", "-am", "-pl", "liquigraph-cli", "clean", "package", "-DskipTests"
     (buildpath/"binaries").mkpath
     system "tar", "xzf", "liquigraph-cli/target/liquigraph-cli-bin.tar.gz", "-C", "binaries"
     libexec.install "binaries/liquigraph-cli/liquigraph.sh"
     libexec.install "binaries/liquigraph-cli/liquigraph-cli.jar"
-    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", Language::Java.java_home_env("1.8")
+    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", :JAVA_HOME => Formula["openjdk@11"].opt_prefix
```
Comment (SMillerDev) [inline Formula/liquigraph.rb:26]:
```suggestion
    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", ::JAVA_HOME => "${JAVA_HOME:-#{ENV["JAVA_HOME"]}}"
```

**Review by SMillerDev — COMMENTED** [review]
Since you mentioned it works with Java 13, this would be better

File: Formula/liquigraph.rb [hunk 7]
```diff
@@ -13,17 +13,17 @@ class Liquigraph < Formula
...
-    ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
+    cmd = Language::Java.java_home_cmd("11")
+    ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
     system "mvn", "-B", "-q", "-am", "-pl", "liquigraph-cli", "clean", "package", "-DskipTests"
     (buildpath/"binaries").mkpath
     system "tar", "xzf", "liquigraph-cli/target/liquigraph-cli-bin.tar.gz", "-C", "binaries"
     libexec.install "binaries/liquigraph-cli/liquigraph.sh"
     libexec.install "binaries/liquigraph-cli/liquigraph-cli.jar"
-    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", Language::Java.java_home_env("1.8")
+    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", ::JAVA_HOME => "${JAVA_HOME:-#{ENV["JAVA_HOME"]}}"
```
Comment (bayandin) [inline Formula/liquigraph.rb:26]:
```suggestion
    (bin/"liquigraph").write_env_script libexec/"liquigraph.sh", :JAVA_HOME => "${JAVA_HOME:-#{ENV["JAVA_HOME"]}}"
```

**Review by bayandin — CHANGES_REQUESTED** [review]
And could you please squash all the commits into one with a message `liquigraph 4.0.1`.

Thanks!

**Comment by SMillerDev** [comment]
The test is failing.

**Comment by fbiville** [comment]
@SmillerDev now fixed

**Comment by SMillerDev** [comment]
Thanks @fbiville  Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**